crates/rolldown/src/module_finalizers/scope_hoisting/impl_visit_mut.rs

```diff
@@ -532,7 +538,13 @@ impl<'me, 'ast> VisitMut<'ast> for ScopeHoistingFinalizer<'me, 'ast> {
                   )),
                 );
               }
-              WrapKind::None => {}
+              WrapKind::None => {
+                // The nature of `import()` is to load the module dynamically/lazily, so imported modules would
+                // must be wrapped, so we could make sure the module is executed lazily.
+                if cfg!(debug_assertions) {
```

Comment:
why only `unreachable` in `debug` mode?

Reply:
Don't want panic happen in release mode. And this also keep the code behave like the old code. I haven't sure if we should panic always in such patterns. I could change it once we come up with the conclusion.